frontend/src/app/main-table/resource-table/resource-table.component.ts

```diff
@@ -27,31 +28,31 @@ export class ResourceTableComponent{
   dataSource = new MatTableDataSource();
 
   constructor(
-    private dialog: MatDialog
-  ) {}
+    private dialog: MatDialog,
+    private translate: TranslateService
+  ) { }
 
   // Resource (Notebook) Actions
   connectResource(rsrc: Resource): void {
     window.open(`/notebook/${rsrc.namespace}/${rsrc.name}/`);
   }
-  
+
   deleteResource(rsrc: Resource): void {
     const dialogRef = this.dialog.open(ConfirmDialogComponent, {
       width: "fit-content",
       data: {
-        title: "You are about to delete Notebook Server: " + rsrc.name,
+        title: this.translate.instant("ressourceTable.deleteDialogTitle") + rsrc.name,
         message:
-          "Are you sure you want to delete this Notebook Server? " +
-          "Your data might be lost if the Server is not backed by persistent storage.",
-        yes: "delete",
-        no: "cancel"
+          this.translate.instant("ressourceTable.deleteDialogMessage"),
+        yes: this.translate.instant("ressourceTable.deleteDialogYes"),
+        no: this.translate.instant("ressourceTable.deleteDialogNo")
       }
     });
     dialogRef
       .afterClosed()
       .pipe(first())
       .subscribe(result => {
-        if (result !== "delete") {
+        if (result !== this.translate.instant("ressourceTable.deleteDialogYes")) {
```

Comment:
Usually would say not to depend on a label string like this, but I know it's coming from the upstream architecture. So suggestion: instead of looking up the value twice, store this in a variable before creating `dialogRef` and then reference that in both places it's used.